amqp_channel.c

```diff
@@ -331,7 +336,7 @@ static PHP_METHOD(amqp_channel_class, __construct)
 	amqp_connection_object *connection;
 
 	/* Parse out the method parameters */
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "o", &connection_object) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "O", &connection_object, amqp_connection_class_entry) == FAILURE) {
```

Comment:
Ensure parameter type is checked (warning with php 5 and 7, TypeError exception with php 8)